### PR TITLE
[Spark] Update the Spark library to align with the latest file format design

### PR DESCRIPTION
**Pull request by lixueclaire**
## Proposed changes

In the latest design of GraphAr file format, the *vertex_count* and *edge_count* files are introduced for storing the edges.
This change adds related methods in the GraphAr Spark library for reading/writing *vertex_count* and *edge_count* files.

## Types of changes

What types of changes does your code introduce to GraphAr?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [CONTRIBUTING](https://github.com/alibaba/GraphAr/blob/main/CONTRIBUTING.rst) doc
- [x] I have signed the CLA
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

## Further comments

solve #143

### CODE REVIEW

File: spark/src/main/scala/com/alibaba/graphar/EdgeInfo.scala [hunk 1]
```diff
@@ -282,10 +282,51 @@ class EdgeInfo() {
...
+   *  @param adj_list_type type of adj list structure.
+   *  @return the edge num file path. If edge info not support the adj list type,
+   *          raise an IllegalArgumentException error.
+   */
+  def getEdgesNumPathPrefix(adj_list_type: AdjListType.Value): String = {
+    if (containAdjList(adj_list_type) == false)
+      throw new IllegalArgumentException
+    val str: String = prefix + getAdjListPrefix(adj_list_type) + "edge_count"
+    return str
+  }
```
Comment (acezen) [inline spark/src/main/scala/com/alibaba/graphar/EdgeInfo.scala:309]: I am a little  puzzled when we use this function if we already have function `getEdgesNumFilePath(chunk_index, adj_list_type)`

File: spark/src/main/scala/com/alibaba/graphar/reader/EdgeReader.scala [hunk 2]
```diff
@@ -38,6 +39,43 @@ class EdgeReader(prefix: String,  edgeInfo: EdgeInfo, adjListType: AdjListType.V
...
+    val file_path = prefix + "/" + edgeInfo.getVerticesNumFilePath(adjListType)
+    val number = FileSystem.readValue(file_path, spark.sparkContext.hadoopConfiguration)
+    return number
+  }
+
+  /** Load the chunk number of src/dst vertices. */
+  def readVertexChunkNumber(): Long = {
+    val vertices_number = readVerticesNumber
+    var vertex_chunk_size = edgeInfo.getSrc_chunk_size
+    if (adjListType == AdjListType.ordered_by_dest || adjListType == AdjListType.unordered_by_dest)
```
Comment (acezen) [inline spark/src/main/scala/com/alibaba/graphar/reader/EdgeReader.scala:53]: Please always add `{` `}`  for `if` `for` and `while` code block

File: spark/src/main/scala/com/alibaba/graphar/reader/EdgeReader.scala [hunk 3]
```diff
@@ -38,6 +39,43 @@ class EdgeReader(prefix: String,  edgeInfo: EdgeInfo, adjListType: AdjListType.V
...
+    val file_path = prefix + "/" + edgeInfo.getEdgesNumFilePath(chunk_index, adjListType)
+    val number = FileSystem.readValue(file_path, spark.sparkContext.hadoopConfiguration)
+    return number
+  }
+
+  /** Load the total number of edges for this edge type. */
+  def readEdgesNumber(): Long = {
+    val vertex_chunk_number = readVertexChunkNumber
+    var number: Long  = 0
+    for (i <- 0L until vertex_chunk_number)
```
Comment (acezen) [inline spark/src/main/scala/com/alibaba/graphar/reader/EdgeReader.scala:74]: Ditto

**Review by acezen — APPROVED** [review]
Overall LGTM，please address the code style comments.

File: spark/src/main/scala/com/alibaba/graphar/EdgeInfo.scala (hunk 1 above)
Comment (lixueclaire) [reply spark/src/main/scala/com/alibaba/graphar/EdgeInfo.scala:309]: This function is used for getting the total number of edges without getting the vertex chunk number at first.

Comment (acezen) [reply spark/src/main/scala/com/alibaba/graphar/EdgeInfo.scala:309]: Did we have a `edge_count` file to store the total edge num?

Comment (lixueclaire) [reply spark/src/main/scala/com/alibaba/graphar/EdgeInfo.scala:309]: No, you can refer to the [test data](https://github.com/GraphScope/gar-test/tree/main/ldbc_sample/csv/edge/person_knows_person/ordered_by_source). But the total edge num can be got by summing up.

File: spark/src/main/scala/com/alibaba/graphar/reader/EdgeReader.scala (hunk 2 above)
Comment (lixueclaire) [reply spark/src/main/scala/com/alibaba/graphar/reader/EdgeReader.scala]: fixed

File: spark/src/main/scala/com/alibaba/graphar/reader/EdgeReader.scala (hunk 3 above)
Comment (lixueclaire) [reply spark/src/main/scala/com/alibaba/graphar/reader/EdgeReader.scala]: fiexed

Comment (lixueclaire) [reply spark/src/main/scala/com/alibaba/graphar/reader/EdgeReader.scala]: fixed

File: spark/src/main/scala/com/alibaba/graphar/EdgeInfo.scala (hunk 1 above)
Comment (acezen) [reply spark/src/main/scala/com/alibaba/graphar/EdgeInfo.scala:315]: I'm not get this, if we don't have a `edge_count` file, how can we use this function to get the total number of edges, could you show me some code example?

Comment (lixueclaire) [reply spark/src/main/scala/com/alibaba/graphar/EdgeInfo.scala:315]: val edge_num_path_pattern = new Path(prefix + edge_info.getEdgesNumPathPrefix(adj_list_type) + "*")
    val edge_num_files = fs.globStatus(edge_num_path_pattern)
    val edge_num = edge_num_files.map(file => utils.FileSystem.readValue(file.getPath().toString(), spark.sparkContext.hadoopConfiguration).toInt).sum

Comment (acezen) [reply spark/src/main/scala/com/alibaba/graphar/EdgeInfo.scala:315]: > ```
> val edge_num_path_pattern = new Path(prefix + edge_info.getEdgesNumPathPrefix(adj_list_type) + "*")
> val edge_num_files = fs.globStatus(edge_num_path_pattern)
> val edge_num = edge_num_files.map(file => utils.FileSystem.readValue(file.getPath().toString(), spark.sparkContext.hadoopConfiguration).toInt).sum
> ```

Great, this looks reasonable to me now.

**Comment by acezen** [comment]
LGTM

**Pull request merged**